src/NHibernate/Proxy/DynamicProxy/NHibernateProxyObjectReference.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Runtime.Serialization;
+using System.Security;
+
+namespace NHibernate.Proxy.DynamicProxy
+{
+	[Serializable]
+	public sealed class NHibernateProxyObjectReference : IObjectReference, ISerializable
+	{
+		readonly IProxyFactory _proxyFactory;
+		readonly object _identifier;
+
+		public NHibernateProxyObjectReference(IProxyFactory proxyFactory, object identifier)
+		{
+			_proxyFactory = proxyFactory;
+			_identifier = identifier;
+		}
+
+		NHibernateProxyObjectReference(SerializationInfo info, StreamingContext context)
+		{
+			_proxyFactory = (IProxyFactory) info.GetValue(nameof(_proxyFactory), typeof(IProxyFactory));
+			_identifier = info.GetValue(nameof(_identifier), typeof(object));
+		}
+
+		[SecurityCritical]
+		public object GetRealObject(StreamingContext context)
+		{
+			return _proxyFactory.GetProxy(_identifier, null);
```

Comment:
`null` here is for `ISessionImplementation`. On deserialization, the `PersistanceContext` will re-associate the session with the proxy. The proxy does support having a `null` session.

Reply:
> it may be a breaking changed for a custom `IProxyFactory`

In this codepath there will be no custom `IProxyFactory`. 

The LazyInitializer underneath does support accepting `null`.

So, answer is no.